src/Avalonia.Controls/Primitives/Popup.cs

```diff
@@ -860,22 +860,7 @@ private void SetTemplatedParentAndApplyChildTemplates(IControl? control)
         {
             if (control != null)
             {
-                var templatedParent = TemplatedParent;
-
-                if (control.TemplatedParent == null)
-                {
-                    control.SetValue(TemplatedParentProperty, templatedParent);
-                }
-
-                control.ApplyTemplate();
```

Comment:
@grokys why would we call ApplyTemplate here? TemplatedParent doesn't affect actual child template as far as I can tell. No need to invalidate its template then.

Reply:
I looked at the history for this and it comes from 2015 - I have _no_ idea ;) I think the rule for such old code needs to be - if it doesn't break a test then you can change it. If we later find a breakage we'll add a test.